### PR TITLE
22364 renraku rule to check that a subclass of test case has its name ending with test

### PR DESCRIPTION
**Pull request by LionelAkue**

### CODE REVIEW

File: src/Renraku/ReTestClassNameShouldEndWithTest.class.st [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ReTestClassNameShouldEndWithTest class >> checksClass [
+
+	^ true
+]
+
+{ #category : #running }
+ReTestClassNameShouldEndWithTest >> basicCheck: aClass [
+
+
+	^ (aClass isKindOf: TestCase class) and: [(aClass name asString endsWith: 'Test') not].
```
Comment (gcotelli) [inline src/Renraku/ReTestClassNameShouldEndWithTest.class.st]: You can use `aClass inheritsFrom: TestCase`

File: src/Renraku/ReTestClassNameShouldNotEndWithTests.class.st [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+{ #category : #'testing-interest' }
+ReTestClassNameShouldNotEndWithTests class >> checksClass [
+
+	^ true
+]
+
+{ #category : #running }
+ReTestClassNameShouldNotEndWithTests >> basicCheck: aClass [
+
+	^ (aClass isKindOf: TestCase class) and: [aClass name asString endsWith: 'Tests'].
```
Comment (gcotelli) [inline src/Renraku/ReTestClassNameShouldNotEndWithTests.class.st]: You can use `aClass inheritsFrom: TestCase`

File: src/Renraku/ReTestClassNotInPackageWithTestEndingName.class.st [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+{ #category : #'testing-interest' }
+ReTestClassNotInPackageWithTestEndingName class >> checksClass [
+
+	^ true
+]
+
+{ #category : #running }
+ReTestClassNotInPackageWithTestEndingName >> basicCheck: aClass [
+
+	^ (aClass isKindOf: TestCase class) and: [(aClass package name asString endsWith: '-Tests') not].
```
Comment (gcotelli) [inline src/Renraku/ReTestClassNotInPackageWithTestEndingName.class.st]: You can use `aClass inheritsFrom: TestCase`

**Comment by LionelAkue** [comment]
Thank you. I get a look.

**Comment by Ducasse** [comment]
Failing tests do not seem related to this PR so I will integrate it.

**Pull request merged**